landoapi/transplants.py

```diff
@@ -495,43 +606,330 @@ def check_landing_blockers(
+                assessment.warnings.append(reason)
+
+    return assessment
+
+
+def assess_transplant_request(
+    phab: PhabricatorClient,
+    supported_repos: dict[str, Repo],
     stack_data: RevisionData,
+    stack: RevisionStack,
+    relman_group_phid: str,
     data_policy_review_phid: str,
-):
-    """Build all transplant blocker checks that need extra Phabricator data"""
-    assert all((isinstance(r, Repo) for r in repositories.values()))
-
-    return DEFAULT_OTHER_BLOCKER_CHECKS + [
-        # Configure uplift check with extra data.
-        check_uplift_approval(relman_group_phid, repositories, stack_data),
-        # Configure data policy check with the project PHID.
-        check_revision_data_classification(data_policy_review_phid),
-    ]
+    landing_assessment: Optional[LandingAssessmentState] = None,
+) -> tuple[TransplantAssessment, TransplantAssessmentState]:
+    """Assess the transplant request."""
+    # Get data from Lando.
+    landable_repos = get_landable_repos_for_revision_data(stack_data, supported_repos)
+
+    involved_phids = set()
+    reviewers = {}
+    for revision in stack_data.revisions.values():
+        involved_phids.update(gather_involved_phids(revision))
+        reviewers[revision["phid"]] = get_collated_reviewers(revision)
+
+    # Get more Phabricator data.
+    involved_phids = list(involved_phids)
+    users = user_search(phab, involved_phids)
+    projects = project_search(phab, involved_phids)
+
+    secure_project_phid = get_secure_project_phid(phab)
+    testing_tag_project_phids = get_testing_tag_project_phids(phab)
+    testing_policy_phid = get_testing_policy_phid(phab)
+
+    # Create a copy of the stack where we will remove revisions that are blocked from
+    # landing, leaving a graph where each path is landable.
+    landable_stack = copy.deepcopy(stack)
+
+    # We only want to consider paths starting from the open revisions
+    # do grab the status for all revisions.
+    statuses = {
+        phid: PhabricatorRevisionStatus.from_status(
+            PhabricatorClient.expect(revision, "fields", "status", "value")
+        )
+        for phid, revision in stack_data.revisions.items()
+    }
+
+    transplant_state = TransplantAssessmentState(
+        phab=phab,
+        stack_data=stack_data,
+        stack=stack,
+        statuses=statuses,
+        landable_stack=landable_stack,
+        landable_repos=landable_repos,
+        supported_repos=supported_repos,
+        reviewers=reviewers,
+        users=users,
+        projects=projects,
+        data_policy_review_phid=data_policy_review_phid,
+        relman_group_phid=relman_group_phid,
+        secure_project_phid=secure_project_phid,
+        testing_tag_project_phids=testing_tag_project_phids,
+        testing_policy_phid=testing_policy_phid,
+        landing_assessment=landing_assessment,
+    )
+
+    # Where we check the landing blockers.
+    assessment = check_landing_lints(transplant_state)
```

Comment:
Referring to things as blockers and lints interchangeably here, I think we should pick a more appropriate term and stick with it (e.g., "landing checks" as suggested in other comments).